### PR TITLE
boards: arm: add STM32H735G discovery kit

### PR DESCRIPTION
**Pull request by benediktibk**
Add the STM32H735G discovery kit to the available boards.

Signed-off-by: Benedikt Schmidt <benedikt.schmidt@embedded-solutions.at>

### CODE REVIEW

**Review by FRASTM — COMMENTED** [review]
LGTM
did you already run some basic tests (like 'hello word' and blinky) on this board

File: boards/arm/stm32h735g_disco/Kconfig.board [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# STM32H735G Discovery board configuration
+
+# Copyright (c) 2019 Linaro Limited
```
Comment (FRASTM) [inline boards/arm/stm32h735g_disco/Kconfig.board:3]: with _2021_ ?

**Comment by benediktibk** [comment]
> LGTM
> did you already run some basic tests (like 'hello word' and blinky) on this board

Yes, blinky and hello world run on it. I have our custom application which is doing a little bit more (ethernet, e.g.), but I cannot add this as a sample due to legal reasons.

**Comment by benediktibk** [comment]
image

File: boards/arm/stm32h735g_disco/Kconfig.board (hunk 1 above)
Comment (benediktibk) [reply boards/arm/stm32h735g_disco/Kconfig.board]: Thanks, copy-paste mistake.

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/stm32h735g_disco/Kconfig.board (hunk 1 above)
[the remark this answers is not in the document]
Comment (benediktibk) [reply boards/arm/stm32h735g_disco/Kconfig.board]: Yes, went already through all the new files and updated the copyright. Just forgot it initially.

**Review by erwango — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**